File: src/main/java/com/agentstore/dependency/exception/DependencyCycleDetectedException.java
package com.agentstore.dependency.exception;

import com.agentstore.common.exception.client.ClientException;
import com.agentstore.common.exception.constants.ErrorCode;

public final class DependencyCycleDetectedException extends ClientException {
    public DependencyCycleDetectedException(String cyclePath) {
        super(ErrorCode.DEPENDENCY_CYCLE_DETECTED, cyclePath);
    }
}

File: src/main/kotlin/com/agentstore/dependency/exception/DependencyCycleDetectedException.kt
package com.agentstore.dependency.exception

import com.agentstore.common.exception.client.ClientException
import com.agentstore.common.exception.constants.ErrorCode

class DependencyCycleDetectedException(cyclePath: String) : ClientException(
    ErrorCode.DEPENDENCY_CYCLE_DETECTED,
    cyclePath,
)
